### DISCUSSION TITLE
Script to translate a quarto doc and output to a new language

### DISCUSSION THREAD

**Discussion by DOH-FAA3303** [quarto-dev/quarto-cli#10585] [Show and Tell]
### Description

I'm sure someone else has done this, but if not, here's a workflow for translating the language of your quarto doc. I loved what @maelle [created here](https://github.com/quarto-dev/quarto-cli/discussions/5262) and I was hoping this workflow could work in conjunction with the [babelquarto package](https://docs.ropensci.org/babelquarto/) to do the actual translating and creation of the language qmd files.

[here is the blog post summarizing everything](https://edenian-prince.github.io/blog/posts/2024-08-21-translate-md-files/)

# summary

- I made a simple package that uses hugging face transformers where the user can input a pre-trained (**and free**) translation model.
- I'm using the awesome `lightparser` [package](https://thinkr-open.github.io/lightparser/) to parse a qmd
- I translate the text of the qmd using the hugging face transformer
- Then `lightparser` does the work convert the translated dataframe back into a new qmd file

![image](https://github.com/user-attachments/assets/809c93ae-d3f4-4c5c-b55d-8bb426c997a4)
the qmd on the left gets translated to spanish on the right - :fire:

edit- i'm aware that [babeldown](https://docs.ropensci.org/babeldown/) can do the translating for you, but it requires a DeepL api. I wasn't able to utilize their free api without creating an account and adding a credit card. Plus, they have limits to their free api. I wanted to try and simplify things by using hugging face transformers and avoiding any credit card hassle.

**Comment by edenian-prince** [comment]
woops! i made this from the wrong github account and cant recreate the post :(  if you want to collaborate or have a question, this is my personal github account. feel free to message!
